packages/js/product-editor/src/components/feedback-bar/feedback-bar.tsx

```diff
@@ -0,0 +1,203 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import {
+	createElement,
+	createInterpolateElement,
+	Fragment,
+} from '@wordpress/element';
+import { closeSmall } from '@wordpress/icons';
+import { WooFooterItem } from '@woocommerce/admin-layout';
+import { Pill } from '@woocommerce/components';
+import {
+	ALLOW_TRACKING_OPTION_NAME,
+	SHOWN_FOR_ACTIONS_OPTION_NAME,
+	STORE_KEY,
+} from '@woocommerce/customer-effort-score';
+import { OPTIONS_STORE_NAME, Product } from '@woocommerce/data';
+import { recordEvent } from '@woocommerce/tracks';
+
+/**
+ * Internal dependencies
+ */
+import {
+	PRODUCT_EDITOR_SHOW_FEEDBACK_BAR_OPTION_NAME,
+	PRODUCT_EDITOR_FEEDBACK_CES_ACTION,
+} from '../../constants';
+import { useFeedbackBar } from '../../hooks/use-feedback-bar';
+
+export type FeedbackBarProps = {
+	product: Partial< Product >;
+};
+
+export function FeedbackBar( { product }: FeedbackBarProps ) {
+	const { showCesModal, showProductMVPFeedbackModal } =
+		useDispatch( STORE_KEY );
+	const { updateOptions } = useDispatch( OPTIONS_STORE_NAME );
+	const {
+		isFeedbackBarSetToShow,
+		allowTracking,
+		cesShownForActions,
+		wasFeedbackModalPreviouslyShown,
+		resolving: isLoading,
+	} = useSelect( ( select ) => {
+		const { getOption, hasFinishedResolution } =
+			select( OPTIONS_STORE_NAME );
+
+		const showFeedbackBarOption = getOption(
+			PRODUCT_EDITOR_SHOW_FEEDBACK_BAR_OPTION_NAME
+		) as string;
+
+		const shownForActions =
+			( getOption( SHOWN_FOR_ACTIONS_OPTION_NAME ) as string[] ) || [];
+
+		const allowTrackingOption =
+			getOption( ALLOW_TRACKING_OPTION_NAME ) || 'no';
+
+		const resolving =
+			! hasFinishedResolution( 'getOption', [
+				SHOWN_FOR_ACTIONS_OPTION_NAME,
+			] ) ||
+			! hasFinishedResolution( 'getOption', [
+				PRODUCT_EDITOR_SHOW_FEEDBACK_BAR_OPTION_NAME,
+			] ) ||
+			! hasFinishedResolution( 'getOption', [
+				ALLOW_TRACKING_OPTION_NAME,
+			] );
+
+		return {
+			cesShownForActions: shownForActions,
+			wasFeedbackModalPreviouslyShown: shownForActions.includes(
+				PRODUCT_EDITOR_FEEDBACK_CES_ACTION
+			),
+			allowTracking: allowTrackingOption === 'yes',
+			isFeedbackBarSetToShow: showFeedbackBarOption === 'yes',
```

Comment:
Could we rename this to `shouldShowFeedbackBar` and also include the check for `wasFeedbackModalPreviouslyShown` here so that we don't need to know both of these properties?

Might also be nice to move all of this logic to the feedback bar hook (or a new hook if that makes more sense).

Reply:
Done